providers/base/bin/network_device_info.py

```diff
@@ -99,7 +97,13 @@ def get_ipv6_address(interface):
         cmd = ['/sbin/ip', '-6', '-o', 'addr', 'show', 'dev', interface,
                'scope', 'link']
         proc = check_output(cmd, universal_newlines=True)
-        return proc.split()[3].strip()
+        try:
+            ipv6_addr = proc.split()[3].strip()
+        except Exception as e:
+            print("ERROR: getting the IPv6 address for %s: %s" %
```

Comment:
I'm not a fan of this for two reasons:

1. the [`print()` statement](https://docs.python.org/3/library/functions.html?highlight=print#print) will print something to `stdout`. Checkbox handles `stdout` and `stderr` differently: things printed to `stdout` will show up green, while things printed to `stderr` will show up red. In your case, this error (and the other one for ipv4) will show up as a regular output in the job.
2. because of this, the printed error messages also show up when running the test cases:

```
(...)
test_get_ipv4_address_without_connection (test_network_device_info.GetIpv4AddressTests) ... ERROR: getting the IPv4 address for wlo1: OSError('packed IP wrong length for inet_ntoa')
ok
test_get_ipv6_address_with_connection (test_network_device_info.GetIpv6AddressTests) ... ok
test_get_ipv6_address_without_connection (test_network_device_info.GetIpv6AddressTests) ... ERROR: getting the IPv6 address for wlo1: IndexError('list index out of range')
ok
(...)
```

That said, I'm not sure what's the best way to proceed. Maybe @kissiel or @Hook25 would have an idea?

Reply:
So unit test shouldn't shows error if they pass the test?
I think maybe network_device_info.py shouldn't try to get info if the conditions is not suitable, perhaps adding conditional expressions could avoid this happening.

Reply:
If getting an ipv6 is part of the test (as in checkbox job) it should definitely fail the test. If this is only for warning purposes, then make it as such, and as Pierre mentioned, dump it on stderr.
Also, the typical logging template of:
`{LEVEL}: {MSG}`  like 'ERROR: no permission' would suggest that the fact that "getting the IPv6 address for" is the error condition, so I'd rephrase it to: "failed to get... " or whatever you prefer, but the message should communicate what went wrong.

As for the final, and weirdest thing: I completely agree that **UNIT TEST** should not invoke a real "backend" utility, like `ip` in our case.